index.js

```diff
@@ -1,5 +1,11 @@
 /**
  * @typedef {import('unist').Node} Node
+ * @typedef {import('unist').Parent} Parent
+ */
+
+/**
+ * @template {Node} [T = Node]
+ * @typedef {import('./complex-types.js').MapFunction<T>} MapFunction<T>
```

Comment:
```suggestion
 * @typedef {import('./complex-types.js').MapFunction<Tree>} MapFunction
 *   Function called with a node, its index, and its parent to produce a new
 *   node.
```

Some docs.

I tested, and the `<T>` / `<Tree>` at the end *is not used*: it’s output as a comment in the `index.d.ts` file. The `@template` above already does it.